hoomd/mpcd/DryingDropletStreamingMethod.h

```diff
@@ -0,0 +1,271 @@
+ */
+
+class PYBIND11_EXPORT DryingDropletStreamingMethod : public mpcd::ConfinedStreamingMethod<mpcd::detail::SphereGeometry>
+    {
+    public:
+        //! Constructor
+        /*!
+         * \param sysdata MPCD system data
+         * \param cur_timestep Current system timestep
+         * \param period Number of timesteps between collisions
+         * \param phase Phase shift for periodic updates
+         * \param R is the radius of sphere
+         * \param bc is boundary conditions(slip or no-slip)
+         */
+        DryingDropletStreamingMethod(std::shared_ptr<mpcd::SystemData> sysdata,
+                                    unsigned int cur_timestep,
+                                    unsigned int period,
+                                    int phase, std::shared_ptr<::Variant> R, const Scalar density, unsigned int seed, mpcd::detail::boundary bc)
+        : mpcd::ConfinedStreamingMethod<mpcd::detail::SphereGeometry>(sysdata, cur_timestep, period, phase, std::shared_ptr<mpcd::detail::SphereGeometry>()),
+          m_R(R), m_density(density), m_seed(seed), m_picks(m_exec_conf), m_bc(bc), m_geometry_set(true)
+          {}
+
+        //! Implementation of the streaming rule
+        virtual void stream(unsigned int timestep);
+
+    protected:
+        std::shared_ptr<::Variant> m_R; //!Radius of Sphere
+        const Scalar m_density;             //!solvent density
+        unsigned int m_seed;            //!< Seed to evaporator pseudo-random number generator
+        const mpcd::detail::boundary m_bc;
+        bool m_geometry_set;           //!< If true, geom is not initialised at first 
+
+        GPUVector<unsigned char> m_bounced_index;          //! indices of bounced particles
+        GPUVector<unsigned int> m_picks;                   //!particles picked for keep/removing on this rank
+
+        unsigned int m_Npick;
+    private:
+        std::vector<unsigned int> m_all_picks;             //!All picked particles
+
+        //! For Making a random pick of particles across all ranks
+        void makeAllPicks(unsigned int timestep, unsigned int N_pick_total, unsigned int N_bounced_total);
+    };
+
+/*!
+ * \param timestep Current time to stream
+ */
+
+void DryingDropletStreamingMethod::stream(unsigned int timestep)
+    {
+    //compute final Radius and Velocity of surface
+    const Scalar start_R = m_R->getValue(timestep);
+    const Scalar end_R = m_R->getValue(timestep + m_period);
+    const Scalar V = (end_R - start_R)/(m_period * m_mpcd_dt);
+    
+    if (V > 0)
+        {
+        throw std::runtime_error("Droplet radius must decrease.");
+        }
+    
+    /*
+     * If the initial geometry was not set. Set the geometry and validate the geometry 
+     * Because the interface is shrinking, it is sufficient to validate only the first time the geometry
+     * is set.
+     */
+    if (m_geometry_set && m_validate_geom)
```

Comment:
I think you only need to check this, since if the geometry has not been set it cannot have been validated:
```suggestion
    if (!m_geom)
```